vertx-db2-client/src/main/java/io/vertx/db2client/impl/drda/DRDAQueryRequest.java

```diff
@@ -1193,11 +1199,17 @@ private Hashtable computeProtocolTypesAndLengths(
                     break;
                 case Types.BINARY:
                 case Types.VARBINARY:
-                    byte[] ba = (byte[]) inputRow[i];
-                    if (ba == null) {
+//                    byte[] ba = (byte[]) inputRow[i];
+                	int length = -1;
+                	if (inputRow[i] instanceof byte[]) {
+                		length = ((byte[]) inputRow[i]).length;
+                	} else if (inputRow[i] instanceof ByteBuf) {
+                		length = ((ByteBuf) inputRow[i]).readableBytes();
+                	}
+                    if (inputRow[i] == null) {
```

Comment:
If inputRow[i] can be null, should that be checked before lines 1204-107 where it could cause a NPE?  Or is the case where it would be null mutually exclusive from when it would be cast as a byte[] or ByteBuf?

Reply:
good question, I had to double check on this and the statement `if (null instanceof byte[])` evaluates to `false` so we don't need the extra null check